File: integrations/aws-strands/error-code-corpus/typescript/frame_in_constructor.ts
/** A frame built inside a constructor, against a local that shadows the file. */
import { EventType, type BaseEvent } from "@ag-ui/core";

const failureCode = "AT_FILE_SCOPE";

export class Reporter {
  readonly frame: BaseEvent;

  constructor() {
    const failureCode = "IN_CONSTRUCTOR";
    this.frame = {
      type: EventType.RUN_ERROR,
      message: "Built in a constructor.",
      code: failureCode,
    };
  }
}

File: integrations/aws-strands/error-code-corpus/python/frame_in_constructor.py
"""A frame built inside a constructor, against a local that shadows the file."""

from ag_ui.core import EventType, RunErrorEvent

failure_code = "AT_FILE_SCOPE"


class Reporter:
    def __init__(self) -> None:
        failure_code = "IN_CONSTRUCTOR"
        self.frame = RunErrorEvent(
            type=EventType.RUN_ERROR,
            message="Built in a constructor.",
            code=failure_code,
        )
